src/Umbraco.Core/Models/SimpleContentType.cs

```diff
@@ -26,7 +54,11 @@ public SimpleContentType(IContentType contentType)
         public string Alias { get; }
 
         /// <inheritdoc />
-        public int Id { get;  }
+        public int Id
+        {
+            get => _id;
+            set => throw new NotSupportedException();
```

Comment:
Would it be better to Explicitly implement the IUmbracoEntity so these setters are hidden. The ISimpleContentType can expose only the getters and that can be explicitly implemented to, then the dev really wont be tricked to thinking its settable with intellisense. Same goes for IdName,and all the other ones where not implemented is thrown.

Reply:
the setters are on the `IUmbracoEntity` interface so we cannot really be an `IUmbracoEntity` without the setters